src/main/java/codechicken/nei/SubsetWidget.java

```diff
@@ -535,6 +546,64 @@ public static void updateHiddenItems() {
         }
     }
 
+    public static void loadCustomSubsets() {
+        File dir = new File(NEIClientConfig.configDir, "subsets");
+
+        if (!dir.exists() || !dir.isDirectory()) {
+            return;
+        }
+
+        Stream.of(dir.listFiles()).filter(file -> !file.isDirectory()).forEach(file -> {
+            ClientHandler.loadSettingsFile(
+                    "subsets/" + file.getName(),
+                    lines -> parseFile(file.getName(), lines.collect(Collectors.toList())));
+        });
+    }
+
+    private static void parseFile(String resource, List<String> itemStrings) {
+        final JsonParser parser = new JsonParser();
+        final String subsetNamespace = resource.substring(0, resource.lastIndexOf('.'));
+        SubsetTag processedTag = new SubsetTag(subsetNamespace, new ItemStackSet());
+
+        for (String itemStr : itemStrings) {
+
+            try {
+
```

Comment:
Remove some extra lines
```suggestion
            try {
```